hibernate-core/src/test/java/org/hibernate/orm/test/query/hql/ParameterTests.java

```diff
@@ -145,7 +145,7 @@ public void testEmbeddableUseInPredicates() {
 	public void testNullParamValues() {
 		inTransaction(
 				session -> {
-					session.createQuery( "from Person p where p.name.first = :p" ).setParameter( "p", null ).list();
```

Comment:
This change and similar others are unnecessary. You should add `NULLS`, `FIRST` and `LAST` to the `identifier` rule in HqlParser.g4 to make these soft keywords.

Reply:
I've done that and the above testing case would fail for all of sudden 'FIRST' and 'LAST' becomes SQL keywords. The above testing case has conflicting 'first' and 'last' columns, thus the code changes.

Reply:
As discussed on Zulip, the `FIRST`, `LAST` and `NULLS` keywords should probably be registered as keywords in the dialects so that matching identifiers are properly quoted

Reply:
Good point. I experimented by adding the three keywords to dialect and found no escaping for 'first' happened. Maybe a missing feature in v6 currently. For that reason, I would maintain the current approach to rename the conflicting columns.